src/device-registry/controllers/create-device.js

```diff
@@ -375,38 +374,30 @@ const device = {
       request["query"]["name"] = name;
       request["body"] = body;
 
-      let responseFromGenerateQRCode = await createDeviceUtil.generateQR(
-        request
-      );
-      // logger.info(
-      //   `responseFromGenerateQRCode -- ${responseFromGenerateQRCode}`
-      // );
-      if (responseFromGenerateQRCode.success === true) {
-        const status = responseFromGenerateQRCode.status
-          ? responseFromGenerateQRCode.status
-          : HTTPStatus.OK;
-        return res.status(status).json({
-          success: true,
-          message: responseFromGenerateQRCode.message,
-          data: responseFromGenerateQRCode.data,
-        });
-      } else if (responseFromGenerateQRCode.success === false) {
-        const status = responseFromGenerateQRCode.status
-          ? responseFromGenerateQRCode.status
-          : HTTPStatus.INTERNAL_SERVER_ERROR;
-        return res.status(status).json({
-          success: false,
-          message: responseFromGenerateQRCode.message,
-          errors: responseFromGenerateQRCode.error
-            ? responseFromGenerateQRCode.error
-            : { message: "" },
-        });
-      }
+      await createDeviceUtil.generateQR(request, (response) => {
+        if (response.success === true) {
```

Comment:
consuming the callback function accordingly. @BenjaminSsempala 

Again, functionality unchanged, just refactoring.

I wonder how the unit tests for this will look like? :)

Reply:
I'll be getting into the tests for controllers soon enough

Reply:
Alright, thanks @BenjaminSsempala